crates/matrix-sdk-ui/src/timeline/read_receipts.rs

```diff
@@ -317,7 +317,7 @@ impl TimelineInnerStateTransaction<'_> {
                 }
 
                 for (user_id, receipt) in receipts {
-                    if receipt.thread != ReceiptThread::Unthreaded {
+                    if !matches!(receipt.thread, ReceiptThread::Unthreaded | ReceiptThread::Main) {
```

Comment:
Is it possible that there's both a threaded-main receipt as well as an unthreaded receipt, for the same user? I think the logic in `maybe_update_read_receipt` would make sure that we only get the most recent, is my understanding correct here?

Reply:
Indeed, it is possible to have both (e.g. a user using both Element Web and EX), but we always only keep the most recent receipt, whether it's explicit in the main thread or unthreaded, or implicit.